gammapy/maps/axes.py

```diff
@@ -2380,13 +2382,16 @@ def format_plot_xaxis(self, ax):
 
         if self.time_format == "iso":
             ax.xaxis.set_major_formatter(DateFormatter("%Y-%m-%d %H:%M:%S"))
-            plt.setp(
-                ax.xaxis.get_majorticklabels(),
-                rotation=30,
-                ha="right",
-                rotation_mode="anchor",
-            )
-
+        elif self.time_format == "mjd":
+            ax.xaxis.set_major_formatter(ticker.StrMethodFormatter("{x:,.5f}"))
```

Comment:
I find the output not so easy to read. Isn't the standard labeling of matplotlib sufficient for this case? 
Maybe the slanted labeling is not necessary as well.

Reply:
If I remove if, the output seems to me not very nice:

![Capture d’écran du 2023-04-17 21-52-51](https://user-images.githubusercontent.com/16781593/232597279-87df6318-fa64-4773-9e09-e386b72d144e.png)

I prefer this one:

![Capture d’écran du 2023-04-17 22-02-18](https://user-images.githubusercontent.com/16781593/232597615-7598160f-9ee8-4c3d-b504-d4d03cd3e21b.png)

This is a question of taster. I leave you decide...
In any case, I have simplified the formatting. Thanks